controllers/space/space_controller.go

```diff
@@ -309,6 +323,33 @@ func NewNSTemplateSetSpec(tmplTier *toolchainv1alpha1.NSTemplateTier) toolchainv
 			s.Namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace(ns)
 		}
 	}
+	// space roles
+	if len(bindings) > 0 {
+		s.SpaceRoles = make([]toolchainv1alpha1.NSTemplateSetSpaceRole, 0, len(tmplTier.Spec.SpaceRoles))
+		// append by alphabetical order of role names
+		roles := make([]string, 0, len(tmplTier.Spec.SpaceRoles))
+		for r := range tmplTier.Spec.SpaceRoles {
+			roles = append(roles, r)
+		}
+		sort.Strings(roles)
+		for _, r := range roles {
+			sr := tmplTier.Spec.SpaceRoles[r]
+			usernames := []string{}
+			for _, b := range bindings {
+				if b.Spec.SpaceRole == r {
+					usernames = append(usernames, b.Spec.MasterUserRecord)
+				}
+			}
+			// no need to add an entry in space roles if there is no associated user
+			if len(usernames) > 0 {
+				s.SpaceRoles = append(s.SpaceRoles, toolchainv1alpha1.NSTemplateSetSpaceRole{
```

Comment:
shouldn't we sort also the usernames so the order is always the same? 
```suggestion
				sort.Strings(usernames)
				s.SpaceRoles = append(s.SpaceRoles, toolchainv1alpha1.NSTemplateSetSpaceRole{
```

Reply:
good point!

Reply:
done in 144bef3